### ISSUE TITLE
Linter should not fail on missing claimNames for XRDs that are designed to be XR only

### ISSUE THREAD

**Issue by ytsarev** [crossplane-contrib/crossplane-lint#1]
I've tried to execute linter against https://github.com/upbound/platform-ref-aws/ and got the following feedback

```
dist/crossplane-lint_linux_amd64_v1/crossplane-lint package -f /home/xnull/upbound/platform-ref-aws/package
crossplane-lint: error: failed to register package schemas: failed to build claim CRD: invalid resource claim names: missing names
```

Related code: https://github.com/crossplane-contrib/crossplane-lint/blob/main/internal/xpkg/lint/schema/xcrd.go#L191-L194

Apparently linter is very opinionated in the expectation that every XRD should have associated Claim and inherently `claimNames`.

At the same time XR-only XRDs without `claimNames` defined are totally valid scenario for the cases like
* https://github.com/upbound/platform-ref-aws/tree/main/package/cluster/network XRD which is designed to be non-namespaced and cluster-wide
* https://github.com/upbound/platform-ref-aws/blob/main/package/cluster/eks/definition.yaml which acts as a nested XR and eventually used in higher-level Composition https://github.com/upbound/platform-ref-aws/blob/main/package/cluster/composition.yaml

**Comment by MisterMX** [comment]
Thanks for the hint.

> Apparently linter is very opinionated in the expectation that every XRD should have associated Claim and inherently claimNames.

Actually, that's not the case. I actually assumed `claimNames` is a required field. Don't ask me why.

Since you are right and it is not the case, this requirement can be removed.

@ytsarev can you explain how CRD generation is handled when `spec.claimNames` is not defined? It appears no claim CRD is generated at all. Is that correct?

**Comment by ytsarev** [comment]
That's totally right, if no claimNames defined then only cluster scoped XR is going to be generated and namespace scoped Claim will not exist.

**Comment by Mitsuwa** [comment]
I opened a pr to try to address this https://github.com/crossplane-contrib/crossplane-lint/pull/7

**Comment by MisterMX** [comment]
Fixed in #7.
